<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>StarkSettingsDialog</class>
 <widget class="QDialog" name="StarkSettingsDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>500</width>
    <height>400</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>STARK Proof Settings</string>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QLabel" name="titleLabel">
     <property name="font">
      <font>
       <family>Poppins</family>
       <pointsize>16</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>STARK Proof Configuration</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="autoGenerationGroup">
     <property name="title">
      <string>Automatic Proof Generation</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QCheckBox" name="enableStarkProofCheckBox">
        <property name="text">
         <string>Enable STARK proof generation</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="autoGenerateProofsCheckBox">
        <property name="text">
         <string>Automatically generate proofs after burn transactions</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="recipientGroup">
     <property name="title">
      <string>HEAT Token Recipient</string>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="recipientLabel">
        <property name="text">
         <string>Default Ethereum Address:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="recipientAddressEdit">
        <property name="placeholderText">
         <string>0x742d35Cc6634C0532925a3b8D4C9db96C4b4d8b6</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="eldernodeGroup">
     <property name="title">
      <string>Eldernode Verification</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QCheckBox" name="enableEldernodeVerificationCheckBox">
        <property name="text">
         <string>Enable Eldernode network verification</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLabel" name="timeoutLabel">
          <property name="text">
           <string>Verification timeout (seconds):</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="timeoutSpinBox">
          <property name="minimum">
           <number>30</number>
          </property>
          <property name="maximum">
           <number>1800</number>
          </property>
          <property name="value">
           <number>300</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="storageGroup">
     <property name="title">
      <string>Proof Storage</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QLabel" name="storagePathLabel">
        <property name="text">
         <string>Storage location:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="storagePathEdit">
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="browseButton">
        <property name="text">
         <string>Browse</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="cleanupGroup">
     <property name="title">
      <string>Cleanup</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <widget class="QCheckBox" name="autoCleanupCheckBox">
        <property name="text">
         <string>Automatically delete old proofs (older than</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="cleanupDaysSpinBox">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>365</number>
        </property>
        <property name="value">
         <number>30</number>
        </property>
        <property name="suffix">
         <string> days)</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="okButton">
       <property name="text">
        <string>OK</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="cancelButton">
       <property name="text">
        <string>Cancel</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="resetButton">
       <property name="text">
        <string>Reset to Defaults</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>okButton</sender>
   <signal>clicked()</signal>
   <receiver>StarkSettingsDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cancelButton</sender>
   <signal>clicked()</signal>
   <receiver>StarkSettingsDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>157</x>
     <y>258</y>
    </hint>
    <hint type="destinationlabel">
     <x>248</x>
     <y>254</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
